### ISSUE TITLE
Update Windows installer to use WiX 4 Thmutil RadioButton to allow WSL/Hyper-V selection

### ISSUE THREAD

**Issue by l0rd** [containers/podman#22492]
### Feature request description

_TL;DR podman windows setup.exe should use radio buttons to present the machine provider alternatives (WSL and HyperV). But that requires WiX v4._

The selection of the machine provider (WSL or Hyper-V) requires radio buttons in the installer GUI. That can be implemented using WiX v3 RadioButton for the MSI package (`podman.msi`) but requires WiX v4 for the `podman-setup.exe` bundle.

In fact the windows installer `podman-setup.exe` is a bundle that includes:
-  the MSI package `podman.msi`
-  the executable `podman-wslkerninst.exe`

Most importantly for this issue, the the `podman-setup.exe` GUI is implemented with [WiX Standard Bootstrapper Application customized using a ThemeFile](https://wixtoolset.org/docs/v3/bundle/wixstdba/wixstdba_customize/) and is different from the `podman.msi` GUI.

[WiX v3 Boostrapper Application Theme](https://wixtoolset.org/docs/v3/xsd/thmutil/theme/) doesn't have radio buttons but [v4 does](https://wixtoolset.org/docs/schema/thmutil/radiobutton/).

### Suggest potential solution

Use https://wixtoolset.org/docs/schema/thmutil/radiobutton/.

### Have you considered any alternatives?

We could use a checkbox

` [ ] use Hyper-V rather then WSL`

but that's [not ideal](https://learn.microsoft.com/en-us/windows/apps/design/controls/radio-button#is-this-the-right-control).
